feat: add optional dev web server #38

jdriscoll98 opened the pull request:
Hello! This is my attempt at solving #38 . I'm new to contributing to open source in general but I think this repo is awesome and want to help in anyway I can. 

I didn't see a section in the contributing docs for writing test cases. I saw some section for evals but I wasn't sure if that was appropriate. Feel free to point me in the right direction and I'll write some tests for this.

jdriscoll98 commented:
Hi! Yes, I tested this myself on the Magnitude demo repo and it worked great. I've also removed the README changes—thanks for the feedback on that.

Regarding the URL parameter, the main reason I can think of for Playwright having a separate webServer config in addition to the baseUrl is because they allow defining multiple web servers, possibly including separate backend services that need individual health checks.

It might make sense for Magnitude to support an array of webServers as well, now that I think about it. Let me know your thoughts!

anerli commented:
this is more of an edge case but im wondering if we should simple return true instead of `return !!res.ok || res.status >= 200;` as long as the fetch doesn't throw? that way we dont make any assumptions about how the web server is configured, even in case it returns some strange < 200 status code

jdriscoll98 commented:
Hey, 

I've updated the webServer type to accept either a single config or an array of configs. I tested this in the magnitude demo repo by running two servers on different ports, and both were up 👍 

I agree with your suggestion regarding the server status check. I've updated the code to return true as long as the fetch doesn't throw.

Let me know if there's anything else I can do!

jdriscoll98 commented:
Awesome! I'm so glad I could contribute.
